### PR TITLE
a-tiny-JS-world: done first aproach

### PR DESCRIPTION
**Pull request by aymkin**
### a-tiny-JS-world

[Demo](https://aymkin.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/aymkin/a-tiny-JS-world/blob/gh-pages/index.js)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/aymkin/a-tiny-JS-world/index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// ========== OUTPUT ===========
+
+[wooman, man, dog, cat].forEach(inhobitant => {
+  print(
+    `I'am a ${inhobitant.species}, my name is <strong>${inhobitant.name}</strong>.
+    I have ${inhobitant.legs} legs and ${inhobitant.hands} hands.
+    Every day I say: "${inhobitant.saying}".
+    I have friends: ${inhobitant.friends.map(friend => friend).join(', ')}.`
+  );
+});
```
Comment (madmaxWMFU) [inline submissions/aymkin/a-tiny-JS-world/index.js]: Please implement new functions that receive your object and returns info about objects and check properties friends on empty values and returns them.
```javascript
getInfoAboutInhabitant(obj) {} 
getFriends() {}
```
This way we break the code into functions for better readability

Comment (aymkin) [reply submissions/aymkin/a-tiny-JS-world/index.js]: @madmaxWMFU I have done this, does it look better now?

File: submissions/aymkin/a-tiny-JS-world/index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+// ========== OUTPUT ===========
+
+[wooman, man, dog, cat].forEach(inhobitant => {
+  getBio(inhobitant);
+  makeSound(inhobitant);
+  getFriends(inhobitant);
+  print('<--->');
+});
```
Comment (madmaxWMFU) [inline submissions/aymkin/a-tiny-JS-world/index.js]: Almost perfect. You did not understand what I was hinting about. In each object, you have described friends, but if we think globally it is possible that someone else does not have friends. Then there will be an error in the output.

The function `makeSound` is superfluous, you can, if you want, implement it in the interior of each object. 
So each object will have its own method of speaking: 
```javascript
const cat = {
 name: "Benjamin",
 ........
 say: () => "meow-meow",
}
```
The function `getFriends` should be called in the function `getBio` and the result should be returned in that case if there are friends, the check is due to the method [hasOwnProperty](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty).

And the last, the function `getBio` should be called in the function `print` according to the task.

File: submissions/aymkin/a-tiny-JS-world/index.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  print(
+    `I'am a ${inhobitant.species}, my name is <strong>${inhobitant.name}</strong>.
+    I have ${inhobitant.legs} legs and ${inhobitant.hands} hands.
+    Every day I say: "${inhobitant.saying}".`
+  );
+};
+
+const makeSound = (inhobitant) => {
+  print(
+    `Every day I say: "${inhobitant.saying}".`
```
[the remark this answers is not in the document]
Comment (aymkin) [reply submissions/aymkin/a-tiny-JS-world/index.js]: Thank you for noticing that, I got fixed it

**Comment by aymkin** [comment]
@madmaxWMFU , @mfialko , @Roka20012

Dear reviewers, all your notes are taken into consideration. Please have a look for another try of code review.

**Comment by madmaxWMFU** [comment]
Great job, this is exactly what I wanted you to do.  Unfortunately, I do not have the right to merge your job.

File: submissions/aymkin/a-tiny-JS-world/index.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  if (inhobitant.hasOwnProperty('friends')) {
+    return `I have friends: ${inhobitant.friends
+      .map(friend => friend)
+      .join(', ')}`;
+  } else {
+    return `I have no friends`;
+  }
+};
+
+const getBio = inhobitant => {
```
[the remark this answers is not in the document]
Comment (aymkin) [reply submissions/aymkin/a-tiny-JS-world/index.js]: done

File: submissions/aymkin/a-tiny-JS-world/index.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  name: 'Aymkin',
+  gender: 'male',
+  legs: 2,
+  hands: 2,
+  say: () => 'Honey, I love borscht.'
+};
+
+const getFriends = inhobitant => {
+  if (inhobitant.hasOwnProperty('friends')) {
+    return `I have friends: ${inhobitant.friends
```
[the remark this answers is not in the document]
Comment (aymkin) [reply submissions/aymkin/a-tiny-JS-world/index.js]: done

File: submissions/aymkin/a-tiny-JS-world/index.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  }</strong>.\nI have ${inhobitant.legs} legs and ${
+    inhobitant.hands
+  } hands.${getFriends(inhobitant)}. \nEvery day I say:  ${inhobitant.say()}`;
+};
+
+// ========== OUTPUT ===========
+
+[wooman, man, dog, cat].forEach(inhobitant => {
+  print(getBio(inhobitant));
+});
```
[the remark this answers is not in the document]
Comment (aymkin) [reply submissions/aymkin/a-tiny-JS-world/index.js]: done, thx a lot for advise

**Comment by aymkin** [comment]
@Temu4 all notes are implemented, thank you for participation.

@babayK0 , please merge, if you can.

**Comment by aymkin** [comment]
@babayK0 , please, merge if possible

**Pull request merged**